c10/core/CPUAllocator.cpp

```diff
@@ -55,16 +55,19 @@ void* alloc_cpu(size_t nbytes) {
 #elif defined(_MSC_VER)
   data = _aligned_malloc(nbytes, gAlignment);
 #else
-  int err = posix_memalign(&data, gAlignment, nbytes);
-  if (err != 0) {
-    CAFFE_THROW(
-        "DefaultCPUAllocator: can't allocate memory: you tried to allocate ",
-        nbytes,
-        " bytes. Error code ",
-        err,
-        " (",
-        strerror(err),
-        ")");
+  // Allocate storage with malloc and manually align data pointer
+  // Then stash storage starting address before the first element to free later
+  // This is significantly faster than posix_memalign
+  const size_t align = std::max(gAlignment, sizeof(void*));
+  void* storage = malloc(nbytes + align);
+  if (storage) {
+    auto storage_addr = reinterpret_cast<uintptr_t>(storage);
+    auto data_addr = storage_addr + align;
+    data_addr -= data_addr % align;
+    data = reinterpret_cast<void*>(data_addr);
+    reinterpret_cast<void**>(data_addr)[-1] = storage;
+  } else {
+    data = nullptr;
```

Comment:
why don't we need to throw if allocation fails like we did before?

Reply:
The `CAFFE_ENFORCE` below raises when `data` is `nullptr`.